src/shell/lisa_gen_buildroot_rootfs

```diff
@@ -0,0 +1,85 @@
+#!/bin/bash
+set -eu
+
+BUILDROOT_URI=git://git.busybox.net/buildroot
+BUILDROOT_VERSION=2018.08
+BUILDROOT_DIR=$LISA_HOME/src/buildroot
+BUILDROOT_PATCHES_DIR=$(dirname $0)/buildroot_patches
+BUILDROOT_CONFIG=$(dirname $0)/buildroot_config
+
+function print_usage {
+	echo "Usage: $0 [options]"
+	echo "	options:"
+	echo "		-p: purge buildroot to force a fresh build"
+	echo "		-h: print this help message"
+}
+
+while getopts "hp" opt
+do
+	case $opt in
+		h)
+			print_usage
+			exit 0
+			;;
+		p)
+			rm -rf $BUILDROOT_DIR
+			exit 0
+			;;
+		?)
+			print_usage
+			exit -1
+			;;
+	esac
+done
+
+# Executed command @$1 if file @$2 doesn't exist
+function do_if {
+	if [ ! -e $2 ]; then
+		$1
+		touch $2
+	fi
+}
+
+function br_clone {
+	git clone $BUILDROOT_URI $BUILDROOT_DIR
+}
+
+function br_checkout_version {
+	pushd $BUILDROOT_DIR > /dev/null
+	git checkout -b $BUILDROOT_VERSION $BUILDROOT_VERSION
+	popd > /dev/null
+}
+
+function br_patch {
+	pushd $BUILDROOT_DIR > /dev/null
+	for patch in $(ls $BUILDROOT_PATCHES_DIR)
+	do
+		git am $patch
+	done
+	popd > /dev/null
+}
+
+function br_apply_config {
+	cp $BUILDROOT_CONFIG $BUILDROOT_DIR/.config
+	make -C $BUILDROOT_DIR olddefconfig
+}
+
+
+do_if br_clone $BUILDROOT_DIR/.lisa_cloned
```

Comment:
I find this `do_if` a bit overkilling and also difficult to undestand without going back to the comment.

What about just using this pattern:
```bash
[[  -f "$BUILDROOT_DIR/.lisa_cloned" ]] || br_clone
```
?

Reply:
Can't find where you generate the `.lisa_cloned` token... can't we just use a file which we know existing into the cloned repo, e.g. a makefile.

Reply:
do_if() touches the file that we pass as a paramter - I can try to rework this to make it clearer. I see the [[]] || command && command; a good short hand but IMHO opinion is also unreadable in this case as you'll pack too much into one line;

I'll see what I can do

Reply:
Right, well... I think I was assuming that you always call a function in the `||` case, and in that case you can just move the touch within that function. Actually, if that was the case, perhaps something like this pattern would have worked just fine:
```bash
function fun1()
    GUARDFILE="$BUILDROOT_DIR/<WHATEVER_GUARD_FILE_MAKES_SENSE"
    [[  ! -f "$BUILDROOT_DIR/.touched_file" ]] || return
    touch $GUARDFILE
    # Function business fallowing...
```

But... your new do_once function seems to actually factor off the pattern above in a common function... and I like it more then your previous proposal and mine above.